proposals/0320-clock-instant-date-duration.md

```diff
@@ -131,98 +108,86 @@ public protocol Clock: Sendable {
+As previously stated, instants need to be compared, and might be stored as a key, but only need to define a concept of now, and a way to advance them given a duration. By utilizing a protocol to define an instant, it provides a mechanism in which to use the right storage for the type, but also be type safe with regards to the clock they are intended for. 
 
+The primary reasoning that instants are useful is that they can be composed. Given a function with a deadline as an instant, if it calls another function that takes a deadline as an instant, the original can just be passed without mutation to the next function. That means that the instant in which that deadline elapses does not have interference with the pre-existing calls or execution time in-between functions. One common example of this is the timeout associated with url requests; a timeout does not fully encapsulate how the execution deadline occurs; there is a deadline to meet for the connection to be established, data to be sent, and a response to be received; a timeout spanning all of those must then have measurement to account for each step, whereas a deadline is static throughout.
 
 ```swift
 public protocol InstantProtocol: Comparable, Hashable, Sendable {
-  func advanced(by duration: Duration) -> Self
-  func duration(to other: Self) -> Duration
+  associatedtype Interval: DurationProtocol
+  func advanced(by duration: Interval) -> Self
+  func duration(to other: Self) -> Interval
 }
 
 extension InstantProtocol {
-  public static func + (_ lhs: Self, _ rhs: Duration) -> Self
-  public static func - (_ lhs: Self, _ rhs: Duration) -> Self
+  public static func + (_ lhs: Self, _ rhs: Interval) -> Self
+  public static func - (_ lhs: Self, _ rhs: Interval) -> Self
   
-  public static func += (_ lhs: inout Self, _ rhs: Duration)
-  public static func -= (_ lhs: inout Self, _ rhs: Duration)
+  public static func += (_ lhs: inout Self, _ rhs: Interval)
+  public static func -= (_ lhs: inout Self, _ rhs: Interval)
   
-  public static func - (_ lhs: Self, _ rhs: Self) -> Duration
+  public static func - (_ lhs: Self, _ rhs: Self) -> Interval
 }
 ```
 
-`InstantProtocol` in addition to the `advance(by:)` and `duration(to:)` methods also has operators to add and subtract durations. However, it does not adhere to `AdditiveArithemtic` since that requires same type addition as well as a "zero"; of which neither make sense generally for defining instants.
+`InstantProtocol` in addition to the `advance(by:)` and `duration(to:)` methods also has operators to add and subtract durations. However, it does not adhere to `AdditiveArithemtic` since that requires same type addition as well as a "zero"; of which neither make sense generally for defining instants. Also it does not adhere to `Strideable` because that requires the stride to be `SignedNumeric` which means that `Duration` would be required to be multiplied by another `Duration` which is inappropriate for two durations. 
```

Comment:
I think the key point isn't that we'd have to provide a `.zero` value ("zero" conceptually makes sense as the epoch for the clock, such as "when the computer was turned on"), but that we'd have to provide a `.zero` value _statically_. While that's possible for some clocks (ones based on some form of computer uptime), it's not possible for _all_ clocks.

Reply:
True, perhaps this can use a bit more clarity in that regards.